pandas/core/indexes/base.py

```diff
@@ -3728,6 +3728,10 @@ def insert(self, loc, item):
         -------
         new_index : Index
         """
+        if lib.checknull(item):
```

Comment:
you can just use ``isna()``

Reply:
The reason I'm using `checknull` over `isna` is to guard against non-scalar `item`.

Using `isna` by itself in an `if` statement would fail for non-scalar `item` with an unhelpful error message, whereas `checknull` returns `False` for any non-scalar:

```python
In [8]: checknull(['a', 'b'])
Out[8]: False

In [9]: bool(isna(['a', 'b']))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-9-358fb1312857> in <module>()
----> 1 bool(isna(['a', 'b']))

ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

The way around this with `isna` would be to do something like: `is_scalar(item) and isna(item)`.  However, looking at the implementation of `isna`, this just forces it down a path to hit `checknull`:

https://github.com/pandas-dev/pandas/blob/cfad581e9a6d35c7d05d2b1f34e4a19f7ee15cc6/pandas/core/dtypes/missing.py#L51-L53

So  `is_scalar(item) and isna(item)` is equivalent to `checknull(item)` but with the additional overhead of two calls to `is_scalar`.

Seems more efficient to just use `checknull`, but can switch to `isna` if that's preferred stylistically.  Or maybe I'm overlooking a better way to using `isna`?

Reply:
As an aside, might be nice to do a `checknull` -> `checkna` renaming in the spirit of `isnull` -> `isna`.

Reply:
yeah should rename that as well.

yeah usually I would just do: ``is_scalar(..) and is_na(...)`` to make this explicit.

Reply:
``lib.checknull`` is pretty internal and don't want to use it in this code